app/WebParsing/ReqParser.hs

```diff
@@ -136,15 +137,17 @@ courseParser = Parsec.between Parsec.spaces Parsec.spaces $ Parsec.choice $ map
 orParser :: Parser Req
 orParser = do
     reqs <- Parsec.sepBy courseParser orSeparator
-    -- TODO: separate cases when reqs has 1 Req vs. multiple Reqs.
-    return $ OR reqs
+    case reqs of
+        [x] -> return x
+        (x:xs) -> return $ OR (x:xs)
```

Comment:
You're getting compiler warnings because the empty list `[]` is not matched by these patterns. In this case, I believe you should use `Parsec`'s `fail` function (it's called something like that) to signal a parsing error. Same issue as below.